test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/TagHelperOutputExtensionsTest.cs

```diff
@@ -968,5 +970,95 @@ public void MergeAttributes_Combines_TagHelperOutputAttributeValues()
             attribute = Assert.Single(tagHelperOutput.Attributes, attr => attr.Name.Equals("for"));
             Assert.Equal(expectedBuilderAttribute.Value, attribute.Value);
         }
+
+        [Fact]
+        public void Single_AddClass()
+        {
+            // Arrange
+            var expectedValue = "class=\"HtmlEncode[[btn]]\"";
+            var htmlEncoder = new HtmlTestEncoder();
+
+            var tagHelperOutput = new TagHelperOutput(
+                tagName: "p",
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(
+                    new DefaultTagHelperContent()));
+
+            // Act
+            tagHelperOutput.AddClass("btn", htmlEncoder);
+
+            // Assert
+            var classAttribute = Assert.Single(tagHelperOutput.Attributes, attr => attr.Name.Equals("class"));
+            Assert.Equal(expectedValue, HtmlContentUtilities.HtmlContentToString(classAttribute));
+        }
+
+        [Fact]
+        public void Multiple_AddClass()
+        {
+            // Arrange
+            var expectedValue = "class=\"HtmlEncode[[btn]] HtmlEncode[[btn-primary]]\"";
+            var htmlEncoder = new HtmlTestEncoder();
+
+            var tagHelperOutput = new TagHelperOutput(
+                tagName: "p",
+                attributes: new TagHelperAttributeList(),
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(
+                    new DefaultTagHelperContent()));
+
+            // Act
+            tagHelperOutput.AddClass("btn", htmlEncoder);
+            tagHelperOutput.AddClass("btn-primary", htmlEncoder);
+
+            // Assert
+            var classAttribute = Assert.Single(tagHelperOutput.Attributes, attr => attr.Name.Equals("class"));
+            Assert.Equal(expectedValue, HtmlContentUtilities.HtmlContentToString(classAttribute));
+        }
+
+        [Fact]
+        public void Single_RemoveClass()
```

Comment:
- Need a new name since this doesn't describe what the test does.
- Add an actual `Single_RemoveClass()` test as well as a `Single_RemoveClass_RemovesEntirely()` test that ends with removing the `class` attribute entirely.
- The behaviour when the value starts with multiple matches may surprise some. I believe a test starting with `HtmlEncode[[btn]] HtmlEncode[[btn]]` in the `class` attribute (obviously not reached using `AddClass(...)` that calls `value.RemoveClass("btn", ...)` will end with a `class` attribute containing `HtmlString(string.Empty)`. Not requiring a change in the product code (up to you) but please demonstrate what happens in a test of this scenario.

Reply:
About this last case. I feel like if the current class value is `HtmlEncode[[btn]] HtmlEncode[[btn]]` and you do `RemoveClass("btn", ...)` then it should remove both and also delete the class attribute itself.

Reply:
Up to you. Add a tests whether or not you change code under src.